File: src/test/testData/java/ExceptionStatements.java
class Tests {
    @Complexity(1)
    void tryDoesNotAddToComplexityNorNesting() {
        try {
            if (true) {                    // +1
                parseFile("salary.txt");
            }
        }
    }

    @Complexity(4)
    void catchAddsToBoth() {
        try {
            parseFile("salary.txt");
        } catch (RuntimeException e) {       // +1
            if (e.message == "Not found") {  // +2 (nesting=1)
                log.warn(e.getMessage())
            }
        } catch (Exception e) {              // +1
            log.error(e.getMessage())
        }
    }

    @Complexity(1)
    void catchWithMultipleExceptions() {
        try {
            rethrow("abc");
        } catch (FirstException | SecondException | ThirdException e) {  // +1
            //below assignment will throw compile time exception since e is final
            //e = new Exception();
            System.out.println(e.getMessage());
        }
    }
}

File: src/test/testData/python/exception_statements.py
@complexity(2)
def try_multiple_catch():
    try:
        f = open("demofile.txt")
    except NameError:         # +1
        print("a")
    except:                   # +1
        print("b")
    finally:
        f.close()


@complexity(1)
def try_does_not_add_to_complexity_nor_nesting():
    try:
        if True:                        # +1
            parseFile("salary.txt")


@complexity(3)
def except_adds_to_both():
    try:
        f = open("demofile.txt")
    except:                   # +1
        if a:                 # +2
            print("a")
    finally:
        f.close()
